### PR TITLE
f-fix/gh-1014-represent-time-in-days

### PR DESCRIPTION
**Pull request by jprotic**
Resolves: #1014

### CODE REVIEW

File: src/main/frontend/src/components/audit/helpers/getReadableChangeValue.js [hunk 1]
```diff
@@ -33,13 +33,14 @@ export default (field, payload) => {
...
     case 'manualAdjust':
-      return `${payload} hours`;
+      return `${payload / 8} ${payload / 8 === 1 ? 'day' : 'days'}`;
+
     case 'trainingManualAdjust':
-      return `${payload} hours`;
+      return `${payload / 8} ${payload / 8 === 1 ? 'day' : 'days'}`;
     case 'bonusManualAdjust':
-      return `${payload} hours`;
+      return `${payload / 8} ${payload / 8 === 1 ? 'day' : 'days'}`;
```
Comment (iscekic) [inline src/main/frontend/src/components/audit/helpers/getReadableChangeValue.js]: This can potentially show lots of decimals, lets somehow make it only show 2. 😄

Comment (iscekic) [reply src/main/frontend/src/components/audit/helpers/getReadableChangeValue.js]: Similar with all the other places where we divide - multiplication is ok.

File: src/main/frontend/src/components/audit/helpers/getReadableChangeValue.js [hunk 2]
```diff
@@ -33,14 +34,14 @@ export default (field, payload) => {
     case 'comment':
       return payload;
     case 'manualAdjust':
-      return `${payload / 8} ${payload / 8 === 1 ? 'day' : 'days'}`;
+      return `${number(payload / 8)} ${payload / 8 === 1 ? 'day' : 'days'}`;
```
Comment (iscekic) [inline src/main/frontend/src/components/audit/helpers/getReadableChangeValue.js]: Let's make this shorter:

```js
// example
case 'xx':
case 'yy':
  return payload;
case 'zz':
case 'bb':
case 'cc':
  return whatever;
```

File: src/main/frontend/src/pages/administration/helpers.js [hunk 3]
```diff
@@ -22,3 +22,6 @@ export const calculatePage = (employeesTab, employeePage, page) => {
 
   return employeePage + 1;
 };
+
+export const number = value =>
```
Comment (iscekic) [inline src/main/frontend/src/pages/administration/helpers.js]: Interesting function, let's name it better.

File: src/main/frontend/src/store/sagas/leaveProfileSaga.js [hunk 4]
```diff
@@ -34,6 +34,13 @@ import { setShouldTableScrollToTop } from '../actions/administrationActions';
 const getLeaveProfileSaga = function*({ payload }) {
   const leaveProfile = yield call(getLeaveProfileApi(payload));
 
+  leaveProfile.entitlement = leaveProfile.entitlement / 8;
```
Comment (iscekic) [inline src/main/frontend/src/store/sagas/leaveProfileSaga.js]: Let's create copies instead of mutating (throughout this file).

File: src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js [hunk 5]
```diff
@@ -392,28 +401,21 @@ class ManualAdjust extends Component {
...
-                                this.tryParse(values.bonusManualAdjust) +
-                                this.tryParse(values.carriedOver) +
-                                this.tryParse(values.manualAdjust) -
-                                this.tryParse(takenAnnual) -
-                                this.tryParse(pendingAnnual)
-                            ) / 8
-                          }
+                          value={number(
+                            annual +
+                              Number(values.bonusManualAdjust) +
```
Comment (iscekic) [inline src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js]: This can fail if we pass a non-number, eg. `undefined`. Or did you mean to use `number`?

File: src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js [hunk 6]
```diff
@@ -61,26 +70,44 @@ class ManualAdjust extends Component {
...
+      obj.carriedOver = Number(obj.carriedOver) * 8;
+      obj.manualAdjust = Number(obj.manualAdjust) * 8;
+      obj.trainingManualAdjust = Number(obj.trainingManualAdjust) * 8;
+
+      return obj;
+    };
+
+    if (!allowance) return null;
+
+    allowance.annual = allowance.annual / 8;
```
Comment (iscekic) [inline src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js]: Multiple things I dislike about this:

* Mutating props is an offence punishable by law 😄 
* Inline function defined on each render
* Lots of repetition

If this component needs data to be in a certain shape (eg. in this case divided by 8), either do it in the component above or in a lifecycle hook.

For the repetition, since we're doing the same thing but with different property names, we can make this much cleaner by abusing javascript stringly-typed-ness. 💃

File: src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js (hunk 5 above)
Comment (jprotic) [reply src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js]: Yes, that field is only for numbers.

File: src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js [hunk 7]
```diff
@@ -83,28 +68,46 @@ class ManualAdjust extends Component {
...
+      'carriedOver',
+      'manualAdjust',
+      'trainingManualAdjust',
+      'pendingTraining',
+      'pendingAnnual',
+      'takenAnnual',
+      'takenTraining'
+    ];
+
+    typeOfAllowance.forEach(item => (allowance[item] = allowance[item] / 8));
```
Comment (iscekic) [inline src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js]: Let's not modify the argument.

File: src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js [hunk 8]
```diff
@@ -83,28 +68,46 @@ class ManualAdjust extends Component {
...
+
+    return allowance;
   };
 
+  closeDrawer = () => this.props.closeDrawer();
```
Comment (iscekic) [inline src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js]: Why is this needed instead of calling `this.props.closeDrawer` directly?

File: src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js [hunk 9]
```diff
@@ -83,28 +68,46 @@ class ManualAdjust extends Component {
...
+
   showErrorToast = error => this.props.toastrError(error);
 
   render() {
-    if (!this.props.allowance) return null;
+    const { allowance } = this.props;
+    const allowanceInDays = { ...allowance };
+
+    this.allowanceInDays = this.showAllowanceInDays(allowanceInDays);
```
Comment (iscekic) [inline src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js]: Don't assign `allowanceInDays` to a property.

File: src/main/frontend/src/store/sagas/leaveProfileSaga.js [hunk 10]
```diff
@@ -64,7 +81,16 @@ const createLeaveProfileSaga = function*({ payload }) {
 };
 
 const updateLeaveProfileSaga = function*({ payload }) {
-  yield call(updateLeaveProfileApi(payload));
+  const payloadInDays = { ...payload };
+
+  payloadInDays.entitlement = (payloadInDays.entitlement * 8).toString();
```
Comment (iscekic) [inline src/main/frontend/src/store/sagas/leaveProfileSaga.js]: Why do we need `toString` here?

File: src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js [hunk 11]
```diff
@@ -83,28 +68,46 @@ class ManualAdjust extends Component {
     });
   };
 
-  closeDrawer = () => this.props.closeDrawer();
+  stringToNumber = obj => {
```
Comment (iscekic) [inline src/main/frontend/src/pages/administration/components/manual-adjust/ManualAdjust.js]: Bad naming.

File: src/main/frontend/src/store/sagas/leaveProfileSaga.js (hunk 10 above)
Comment (iscekic) [reply src/main/frontend/src/store/sagas/leaveProfileSaga.js]: Make an issue for the BE guys to fix this and leave a `TODO` here. 😄
